File: server/tests/fixtures/alpha_rule_triggers/csharp/arch_data_model.cs
// Should trigger: arch.data_model
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyApp.Models
{
    // Entity Framework data model
    [Table("Users")]
    public class User
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Username { get; set; }

        [Required]
        [EmailAddress]
        public string Email { get; set; }

        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; }
    }

    // Record data model (C# 9+)
    public record Order(string OrderId, int UserId, decimal Amount);
}

File: server/tests/fixtures/alpha_rule_triggers/java/arch_data_model.java
// Should trigger: arch.data_model
package com.example.model;

import javax.persistence.*;

// JPA Entity - data model
@Entity
@Table(name = "users")
public class User {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String username;
    
    @Column(nullable = false, unique = true)
    private String email;
    
    private boolean isActive = true;
    
    // Getters and setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getUsername() { return username; }
    public void setUsername(String username) { this.username = username; }
}

// Record - data model (Java 14+)
record OrderRecord(String orderId, Long userId, double amount) {}
